sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -614,6 +635,7 @@ public OffsetDateTime renewSessionLock() {
      * @throws NullPointerException if {@code sessionId} or {@code maxLockRenewalDuration} is null.
      * @throws IllegalArgumentException if {@code sessionId} is an empty string.
      * @throws IllegalStateException if the receiver is a non-session receiver or the receiver is disposed.
+     * @throws IllegalArgumentException if {@code maxLockRenewalDuration} is negative.
```

Comment:
duplicate IllegalArgumentException 

Also it help to say which type of exception is now mapped to the new exception (added here).

And we probably want to document, what error is to IllegalStateException, what to IllegalArgumentException, what to ServiceBusException, and what to AmqpException.

Reply:
Sure, will change it.

In the old code, there are some methods map the errors to `ServicebusException` , but some don't. These errors which map to `ServicebusException` are occured unexpectedly, such as get link from a disposed session, create session timeout,  most of them are occured by incident which user cannot avoid in their code. I added `mapError()` function to the rest method to wrap these unexpected errors. 

I referred to the method that already used `mapError()`, I have no clue that how could we document what exception map to `ServiceBusException`? If we document all of them in each method, it will be large. Do you have any idea? @conniey @anuchandy @ki1729

Reply:
I would be satisfied with a generic documentation (mostly internally).

E.g. I would prefer we avoid this seemingly conflict error case:
```
     * @throws AmqpException if {@code message} is larger than the maximum size of the {@link
     *     ServiceBusMessageBatch}.
...
    public Mono<Void> sendMessages(Iterable<ServiceBusMessage> messages) {
```

```
     * @throws ServiceBusException If the messages could not be scheduled or the {@code message} is larger than
     *      the maximum size of the {@link ServiceBusMessageBatch}.
...
    public Flux<Long> scheduleMessages(Iterable<ServiceBusMessage> messages, OffsetDateTime scheduledEnqueueTime,```

Reply:
I also agree. I think we can use this opportunity to develop a standard for how to handle different exceptions and how to document them. I not sure if we already have some samples?

Reply:
I haven't found any samples.

Calling a public API,
the static check almost only throw three kind of exception:
- NullPointerException for parameter is null.
- IllegalStateException for client dispose.
- IllegalArgumentException for parameter is empty.

when we call send (or some other method to connect with service) there may be several RuntimeException:
- AmqpException for create a link from a disposed session or wait timeout to get a session
- IOExcpetion for if dispatcher has some errors to close.

and so on. these exception we wrap as a ServiceBusException, I think it's good because these exceptions only occur by incident.

I prefer to wrap AmqpException to ServiceBusException as well.